File: CPP/Program/Headset.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "OutputDevice.cpp"
#include "InputDevice.cpp"

class Headset : public OutputDevice, public InputDevice {
    private:
        bool isWireless;
        int soundLevelDB;
    public:
        Headset() {

        }

        Headset(bool isWireless, int soundLevelDB, string jenisOutput, string OutputLatency, string jenisInput, string inputLatency, string merk, string nama) : 
            OutputDevice(jenisOutput, OutputLatency, merk, nama), 
            InputDevice(jenisInput, inputLatency, merk, nama) {
            this->isWireless = isWireless;
            this->soundLevelDB = soundLevelDB;
        }

        void setIsWireless(bool isWireless) {
            this->isWireless = isWireless;
        }

        void setSoundLevelDB(int soundLevelDB) {
            this->soundLevelDB = soundLevelDB;
        }

        bool getIsWireless() {
            return this->isWireless;
        }

        int getSoundLevelDB() {
            return this->soundLevelDB;
        }

        ~Headset() {

        }

};

File: PYTHON/Program/Headset.py
from OutputDevice import OutputDevice
from InputDevice import InputDevice

class Headset(OutputDevice, InputDevice):
    def __init__(self, isWireless=False, soundLevelDB=0, jenisOutput="", outputLatency="", jenisInput="", inputLatency="", merk="", nama=""):
        OutputDevice.__init__(self, jenisOutput, outputLatency, merk, nama)
        InputDevice.__init__(self, jenisInput, inputLatency, merk, nama)
        self.isWireless = isWireless
        self.soundLevelDB = soundLevelDB

    def setIsWireless(self, isWireless):
        self.isWireless = isWireless

    def setSoundLevelDB(self, soundLevelDB):
        self.soundLevelDB = soundLevelDB

    def getIsWireless(self):
        return self.isWireless

    def getSoundLevelDB(self):
        return self.soundLevelDB
